### PR TITLE
fix: support android

### PR DESCRIPTION
**Pull request by HashEngineering**
Make changes to build.rs to support android builds.

### CODE REVIEW

File: rust-bindings/bls-dash-sys/build.rs [hunk 1]
```diff
@@ -127,7 +164,8 @@ fn main() {
     cc.files(cpp_files)
         .includes(&include_paths)
         .cpp(true)
-        .flag_if_supported("-std=c++14");
+        .flag_if_supported("-std=c++14")
+        .target(&env::var("TARGET").unwrap());
```
Comment (HashEngineering) [inline rust-bindings/bls-dash-sys/build.rs:168]: the `cc` crate does support using the correct compiler based on the `CC_*` and `AR_*` environment variables, but it needs to know the target.

**Review by shumkov — DISMISSED** [review]
👍

**Review by ogabrielides — APPROVED** [review]
utACK

**Review by UdjinM6 — DISMISSED** [review]
utACK

**Review by ogabrielides — APPROVED** [review]
re-utACK

**Review by UdjinM6 — APPROVED** [review]
:)

**Comment by UdjinM6** [comment]
glad we have gh rules that protected from mistakes like that

**Pull request merged**